tests/test_init.py

```diff
@@ -26,3 +26,208 @@ def test_systemd_present(container):
+    reason="Running systemd in docker is broken as of systemd 248, see https://github.com/moby/moby/issues/42275",
+)
+def test_systemd_detect_virt(container, host, container_runtime):
+    """
+    Check that systemd-detect-virt correctly detects the used container runtime
+    """
+    output = container.connection.run_expect([0], "systemd-detect-virt")
+    runtime = container_runtime.runner_binary
+    assert (
+        runtime in output.stdout
+    ), f"systemd-detect-virt failed to detect {runtime}"
+
+
+@pytest.mark.parametrize("container", [INIT_CONTAINER], indirect=True)
+@pytest.mark.skipif(
+    DOCKER_SELECTED
+    and (int(LOCALHOST.package("systemd").version.split(".")[0]) >= 248),
+    reason="Running systemd in docker is broken as of systemd 248, see https://github.com/moby/moby/issues/42275",
+)
+def test_journald(container):
+    """
+    Check that journald behaves correctly
+    """
+
+    # Check that we reached at least the multiuser target
+    journal = container.connection.run_expect([0], "journalctl --boot -e")
+    assert (
+        "Reached target Multi-User System" in journal.stdout
+    ), f"Multi-User target was not reached"
+
+
+@pytest.mark.parametrize("container", [INIT_CONTAINER], indirect=True)
+@pytest.mark.skipif(
+    DOCKER_SELECTED
+    and (int(LOCALHOST.package("systemd").version.split(".")[0]) >= 248),
+    reason="Running systemd in docker is broken as of systemd 248, see https://github.com/moby/moby/issues/42275",
+)
+def test_hostnamectl(container, host, container_runtime):
+    """
+    Check that hostnamectl prints the expected values
+    """
+
+    # Process the given output into a key:value map, one entry by line
+    def split_values(output, sep=":"):
+        ret = {}
+        for line in output.split("\n"):
+            line = line.strip()
+            i = line.find(sep)
+            if i < 0:
+                continue
+            name, value = line[:i].strip(), line[i + 1 :].strip()
+            ret[name] = value
+        return ret
+
+    hostnamectl = container.connection.run_expect([0], "hostnamectl")
+    # Process the printed values to a string map
+    values = split_values(hostnamectl.stdout)
+    assert values["Chassis"] == "container", f"Chassis mismatch"
+    assert (
+        "SUSE Linux Enterprise Server" in values["Operating System"]
+    ), f"Missing SUSE tag in Operating system"
+    assert (
+        values["Virtualization"] == container_runtime.runner_binary
+    ), f"Virtualization tag mismatch"
+
+
+@pytest.mark.parametrize("container", [INIT_CONTAINER], indirect=True)
+@pytest.mark.skipif(
+    DOCKER_SELECTED
+    and (int(LOCALHOST.package("systemd").version.split(".")[0]) >= 248),
+    reason="Running systemd in docker is broken as of systemd 248, see https://github.com/moby/moby/issues/42275",
+)
+def test_timedatectl(container):
+    """
+    Check that timedatectl can be invoked without issues
+    """
+    container.connection.run_expect([0], "timedatectl")
```

Comment:
Maybe add a check that the time & date are roughly correct?

Reply:
good idea. Added it now.